File: ps/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        int i,j;
        if (strs.size() == 0) return "";
        for (j=0;j<strs[0].size();j++){
            char tmp = strs[0][j];
            for (i=0;i<strs.size();i++) {
                if (strs[i][j] != tmp) return strs[0].substr(0,j);
            }
        }
        return strs[0];
    }
};

File: ps/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        res = []
        numberOfWords = len(strs)
        if numberOfWords == 0:
            return ""
        oneOfLength = len(strs[0])

        for idx in range(oneOfLength):
            for idx2 in range(numberOfWords):
                if len(strs[idx2]) <= idx:
                    return "".join(res)
                if strs[0][idx] != strs[idx2][idx]:
                    return "".join(res)
            res.append(strs[0][idx])

        return "".join(res)